pytheus/backends.py

```diff
@@ -1,10 +1,64 @@
+from abc import ABC, abstractmethod
+import json
+import os
 from threading import Lock
+from typing import Any
 
+from pytheus.exceptions import PytheusException
 
-class LockedValue:
-    """Provides a value protected by a threading Lock."""
+BackendConfig = dict[str, Any]
 
-    def __init__(self) -> None:
+
+class Backend(ABC):
+    def __init__(self, config: BackendConfig):
+        self.config = config  # TODO: We should discuss validation here (jsonschema, pydantic, etc)
```

Comment:
I didn't want to go as far as adding validation for the configuration. If it was jsonschema, this could be a top level property of the backend subclass. Alternatively, we could define a Pydantic validator in the same way.